### PR TITLE
:white_check_mark: Add link checker

### LINKED ISSUE #2049: :sparkles: Add an external link checker
### What needs to be documented?

We need to make sure the links we have to external sites aren't broken. A step in the build process that checked all links would be good.

Bear in mind that GitHub can rate limit some requests so a token might be necessary.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CollierCZ**
## Why

We want to ensure all our links are still valid.
Closes #2049 

## What's changed

- Added [muffet](https://github.com/raviqqe/muffet) as a link checker to crawl and check links.
  - It needs to be rate limited for sites like GitHub, which means it takes a while.
  - It's also flaky, especially with Community sometimes timing out for no reason.
- So moved it to a twice-weekly check that will update #2213 with any broken links found.
- Added an internal link checker to test each PR that's submitted.
  - [Vault](https://docs.platform.sh/configuration/services/vault.html) is excluded because it uses headings from transclusion (so they're marked as missing). If there's a problem, it should be caught in the regular check.
- Fixed links identified by both checks.
- Added checks to CONTRIBUTING.md

### CODE REVIEW

File: docs/src/guides/gatsby/headless/strapi.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ $ yarn add strapi-plugin-graphql
...
+You can then modify [`gatsby-config.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/) to read from the backend Strapi container through the `strapi` relationship defined above to configure the `apiURL` attribute for `gatsby-source-strapi`. Notice that the source plugin requires that you explicitly define the `contentTypes` you would like to retrieve from Strapi. At this point you have not built out the API, and the Content Types `article` and `category` are included to support the [post-install instructions](https://github.com/platformsh-templates/gatsby-strapi/tree/update#user-content-post-install) outlined in the template's README. Adjust these values to fit your current API if your are planning on migrating an existing Strapi repository.
```
Comment (trolologuy) [inline docs/src/guides/gatsby/headless/strapi.md:36]:
```suggestion
You can then modify [`gatsby-config.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/) to read from the backend Strapi container through the `strapi` relationship defined above to configure the `apiURL` attribute for `gatsby-source-strapi`.
Note that the source plugin requires that you explicitly define the `contentTypes` you would like to retrieve from Strapi. 
At this point, you have not built out the API, and the Content Types `article` and `category` are included to support the [post-install instructions](https://github.com/platformsh-templates/gatsby-strapi/tree/update#user-content-post-install) outlined in the template's README.
Adjust these values to fit your current API if your are planning on migrating an existing Strapi repository. 
```

Comment (trolologuy) [reply docs/src/guides/gatsby/headless/strapi.md:36]: The wording is a bit heavy and would need some improvements, adding linebreaks for easier future editing and readability.

File: docs/src/integrations/source/bitbucket.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ description: |
 
 It's possible to integrate a Platform.sh project with either the freely available Bitbucket Cloud product
 or with the self-hosted [Bitbucket Server](https://confluence.atlassian.com/bitbucketserver/).
-In both cases, you need to [install the Platform.sh CLI](/development/cli/_index.md#installation), if you haven't already done so,
-to set up the integration.
+In both cases, you need to [install the Platform.sh CLI](/development/cli/_index.md),
```
Comment (trolologuy) [inline docs/src/integrations/source/bitbucket.md:11]: Locally this one doesn't work for me. But [online it seems to be working](https://pr-2189-odiepqq-652soceglkw4u.eu-3.platformsh.site/integrations/source/bitbucket.html),

File: docs/src/integrations/source/github.md [hunk 3]
```diff
@@ -115,13 +115,4 @@ they will be activated automatically with a copy of the parent's data.
...
-
-You can clone your codebase by running `platform get <projectID>`
-or in your project in the console by going to Code > Git and running the `git clone` command.
-
-When you perform this action, you are actually cloning from your remote integrated repository,
-so long as you have the [appropriate access to do so](/administration/users.md#user-access-and-integrations).
-
-Your GitHub repository is considered by Platform.sh to be the "source of truth" for the project.
-The project is only a mirror of that repository, and all commits should be pushed only to GitHub.
+{{% clone-commit name="GitHub" %}}
```
Comment (trolologuy) [inline docs/src/integrations/source/github.md:118]: I just noticed the [Integrations Troubleshoot page](https://pr-2189-odiepqq-652soceglkw4u.eu-3.platformsh.site/integrations/source/troubleshoot.html), should the `{{% troubleshoot %}}` shortcode be added there as well?

**Review by trolologuy — APPROVED** [review]
Great work! That's quite a massive PR. I can't wait to see the automatic link checker live!
A few proposed changes here and there, but nothing big either.

- [x] Final built pages are working as expected.
- [x] The information is correct.
- [x] The information is comprehensive.
- [x] The information is concise.
- [x] The information is clear.
- [x] All links work, including to section headers.

✅ Okey to merge.

File: docs/src/configuration/services/vault.md [hunk 4]
```diff
@@ -36,7 +36,9 @@ You can create multiple endpoints, such as to have key management separate from
 
 ## Use Vault KMS
 
-To connect your app to the Vault KMS, use a token that is defined in the `$PLATFORM_RELATIONSHIPS` environment variable. With this token for authentication, you can use any of the policies you [defined in your `.platform/services.yaml` file](#add-the-service).
+To connect your app to the Vault KMS, use a token that is defined in the `$PLATFORM_RELATIONSHIPS` environment variable.
+With this token for authentication,
+you can use any of the policies you [defined in your `.platform/services.yaml` file](#1-configure-the-service).
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/configuration/services/vault.md:41]: Why do you worry about adding that here in particular? If it is changed to that, I don't think it causes more problems than elsewhere. Or does it?

File: docs/src/development/troubleshoot.md [hunk 5]
```diff
@@ -246,5 +246,5 @@ Cron jobs may invoke other services in unexpected ways, which can increase execu
 
 For language-specific troubleshooting for your apps:
 
-- [PHP](../languages/php/troubleshoot.html)
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/development/troubleshoot.md:249]: I mean, so should I :grin:

File: docs/src/guides/drupal9/solr.md [hunk 6]
```diff
@@ -37,7 +37,7 @@ Once [this issue](https://www.drupal.org/project/drupal/issues/2876675) is resol
 
 If you are using the latest version of Search API Solr and running Drupal 8.8 or higher on PHP 7.2 or higher, this step is no longer necessary. You should be able to proceed directly to the next step.
 
-{{% endpoint-description type="solr" sectionLink="#solr-6-and-later" multipleText="cores" noApp=true /%}}
+{{% endpoint-description type="solr" sectionLink="/configuration/services/solr.html#solr-6-and-later" multipleText="cores" noApp=true /%}}
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/guides/drupal9/solr.md:40]: The service definition link is the same for all services. It's just supposed to be for anyone who doesn't know what a `.platform/services.yaml` file is.

File: docs/src/integrations/source/bitbucket.md (hunk 2 above)
Comment (CollierCZ) [reply docs/src/integrations/source/bitbucket.md:11]: It wasn't relative, which can be fixed.

File: docs/layouts/shortcodes/clone-commit.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+## Clones and commits
+
+You can clone your codebase by running `platform get <projectID>`
+or in your project in the console by going to Code > Git and running the `git clone` command.
+
+When you perform this action, you are actually cloning from your remote integrated repository,
+so long as you have the [appropriate access to do so](/integrations/source/troubleshoot.html).
```
[the remark this answers is not in the document]
Comment (trolologuy) [reply docs/layouts/shortcodes/clone-commit.md]: `If` is indeed even better!

File: docs/src/configuration/services/vault.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (trolologuy) [reply docs/src/configuration/services/vault.md:41]: Hmmm, I'm not sure what I meant here, I should have provided more context. Looks good to me, this comment can be discarded.

**Review by trolologuy — APPROVED** [review]
Thanks for the changes, looks better now 🎉

**Pull request merged**